build/dist/main-application.js

```diff
@@ -279,6 +293,55 @@ export class MainApplication extends LitElement {
     context.stroke();
   }
 
+  _updateInfoPanel(event) {
+    let stats = [{
+      name: 'Event type',
+      value: event.type
+    }, {
+      name: 'Pointer type',
+      value: event.pointerType
+    }, {
+      name: 'Pointer id',
+      value: event.pointerId
+    }, {
+      name: 'Position x',
+      value: this._roundDecimal(event.x, 4)
+    }, {
+      name: 'Position y',
+      value: this._roundDecimal(event.y, 4)
+    }, {
+      name: 'Prefered color',
+      value: event.preferredColor
+    }, {
+      name: 'Pressure',
+      value: this._roundDecimal(event.pressure, 4)
+    }, {
+      name: 'Tangential pressure',
+      value: this._roundDecimal(event.tangentialPressure, 4)
+    }, {
+      name: 'Tilt x',
+      value: this._roundDecimal(event.tiltX, 4)
+    }, {
+      name: 'Tilt y',
+      value: this._roundDecimal(event.tiltY, 4)
+    }, {
+      name: 'Twist',
+      value: this._roundDecimal(event.twist, 4)
+    }];
+    let innerHTML = '';
+
+    for (let i = 0; i < stats.length; i++) {
+      innerHTML += '<tr><th>' + stats[i].name + '</th><td>' + stats[i].value + '</td></tr>';
+    }
+
+    this.shadowRoot.querySelector('#info-panel').innerHTML = innerHTML;
```

Comment:
This could be improved a lot. Rewriting the innerHTML is usually something you should try to avoid. With lit you can use bindings to make this entire block a lot easier. 

Inside your render()/html below (info-panel) write down the structure e.g. <div class="grid"><div id="cell">property: ${this.myproperty}</div>....

Then in _updateInfoPanel() you can update the property by just calling this.myproperty = "test" and the binding will get updated automatically.

See over there https://github.com/foldable-devices/demos/blob/master/battleship/src/enemy-ship.js#L71

Actually I strongly suggest you put the info panel into its own file/component (infopanel.js) and you can use in this render/html fct like this: <info-panel></info-panel> and feed the data to it using properties, e.g. <info-panel x="${this.x}" pressure="{$this.pressure}</info-panel>. This will encapsulate the CSS of the panel and the code into its own file.

Reply:
sure, good to know

Reply:
@darktears Ok, I updated my patch.